### DISCUSSION TITLE
how do I have a static crew and commander in mars-sim ?

### DISCUSSION THREAD

**Discussion by shirishag75** [mars-sim/mars-sim#427] [Q&A]
Is there or would there be a way to start the game with the experience and skills acquired by a team. For e.g. I would like the sim to have an all-ISRO team and the sim runs like it is being run by ISRO only. Is this possible, if yes how ? And by static meaning the same members again and again with the skills leveled up. It also means they do age up and at times die. Also I dunno events such as birthdays, marriage anniversaries, love etc. are part of the dynamic in mars-sim or not. I did see you have personality traits given to the team which people could fiddle with. Not sure if you can see the personality traits in the headless version too.

**Comment by mokun** [comment]
> And by static meaning the same members again and again

I guess you mean you want the option to be able to serialize/save the settlers into a xml/json file and load them later back in any simulation. 

It's definitely more life-like since this would give settlers a sense of continuity. 

Yes we could restructure the core engine to do that :D

**Comment by mokun** [comment]
>  with the skills leveled up. 

At some points we will revamp the experience/skill system since the result of they do on Mars should be more than just `Experience Points` accumulated. 

I hope we could switch to using **Neural Network** topology for data/decision processing and storage of past experiences/decisions. This would make things very interesting.

**Comment by mokun** [comment]
One question is that how you would handle a different version of a same settler ?

What if you can have two settlers having the same name but different level of experiences ? 

One could be a `clone` of another in the past but they have since forked and do different things ? 

We'll need a way to manage how to distinguish who is who.

**Comment by mokun** [comment]
> Also I dunno events such as birthdays, marriage anniversaries, love etc. are part of the dynamic in mars-sim or not.

Yes birthday is there. But we don't have birthday party celebration yet...

We do have a simple model of [Relationship](https://github.com/mars-sim/mars-sim/wiki/Relationship). 

It's basically a `permutation score` of person A toward person B (one-way).

And we keep track of `9 level` of relationship between any two individuals. 

We haven't got to the point of modeling significant life events such as marriage or giving birth to babies yet but it will be inevitable when you have a colony of people not bailing out and choosing to stick together and have gone through things beyond the level of survival... 

>  I did see you have personality traits given to the team which people could fiddle with. Not sure if you can see the personality traits in the headless version too.

The headless edition is using the same core engine as other editions and so have the same features.

If you like, let's open a new thread for sharing/discussing what you want to see or refine on each feature separately ?
